compiler/x/codegen/OMRMemoryReference.cpp

```diff
@@ -1642,14 +1642,33 @@ OMR::X86::MemoryReference::generateBinaryEncoding(
          TR_ASSERT(IS_32BIT_SIGNED(displacement), "64-bit displacement should have been replaced in TR_AMD64MemoryReference::generateBinaryEncoding");
 
          uint8_t divisor = 16;
-         divisor = containingInstruction->getOpCode().info().isEvex256() ? 32 : divisor;
-         divisor = containingInstruction->getOpCode().info().isEvex512() ? 64 : divisor;
+         bool isEvex = true;
```

Comment:
This logic looks identical to that in `case 7`.  Can this be pulled into its own function so we don't have so much code duplication?

Reply:
I have moved the logic up to remove the duplication